DOC: Add note about deprecated offset aliases

tacaswell opened the pull request:
The PRs #55792 (Y), #52064 (Q), and #55553 (M) deprecated the single letter version of the aliases in favour of the -end version of them.

This table was missed as part of that work.

I skipped making an issue because the PR with the fix is the simplest way to express the issue!

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] ~[Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature~
- [ ] ~All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).~
- [ ] ~Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.~
- [ ] ~Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.~

tacaswell committed:
DOC: Add note about alias deprecations

The PRs #55792 (Y), #52064 (Q), and #55553 (M) deprecated the single letter
version of the aliases in favour of the -end version of them.

Add a note to the offset table about deprecations.

MarcoGorelli commented:
thanks @tacaswell 

these are only deprecated for offsets (where people always interpreted `'M'` to mean `'month'` as opposed to `'month end'`)

for periods (the section you've edited), `'M'` is still valid - periods have no "period start" / "period end"

is this not what you're seeing?

tacaswell commented:
I'm getting this out of `pd.Grouper(freq=...)` which in my head should be "period", but reading the docs for `pd.Grouper ` it does link to the offsets table so there is some user-error / wishful-API-usage (🐑 ) going on here.  I went looking for notes on the deprecation, found the `M` in this table and decided to "help". 

The warning I'm getting is:

```
<string>:23: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
```

which is unclear if it is an offset or period (I don't have time to run through sorting out the deprecation note generation to tell if that is easy to add or not, sorry).

-----

I've removed the changes to the table and moved the note to the other table.

MarcoGorelli reviewed (approved):
thanks @tacaswell , should be right - @natmokval could you confirm please?

natmokval commented:
> thanks @tacaswell , should be right - @natmokval could you confirm please?

I agree, we should point out that we deprecated the aliases `'M,' 'Q,' etc.`, for offsets. Thanks, @tacaswell, for adding the note for the table with Offset aliases.

We can also mention in this note that we deprecated the aliases `'BM,' 'CBM,' 'BQ,' and 'BY'` in favor of `'BME,' 'CBME,' 'BQE,' and 'BYE'`, but it’s already clear enough.

MarcoGorelli commented:
agree, thanks for taking a look

cool, let's ship it then
